rasa/core/policies/intent_ted_policy.py

```diff
@@ -0,0 +1,492 @@
+        MAX_POS_SIM: 0.8,
+        # Maximum negative similarity for incorrect labels.
+        # Should be -1.0 < ... < 1.0 for 'cosine' similarity type.
+        MAX_NEG_SIM: -0.2,
+        # If 'True' the algorithm only minimizes maximum similarity over
+        # incorrect intent labels, used only if 'loss_type' is set to 'margin'.
+        USE_MAX_NEG_SIM: True,
+        # If 'True' scale loss inverse proportionally to the confidence
+        # of the correct prediction
+        SCALE_LOSS: False,
+        # ## Regularization parameters
+        # The scale of regularization
+        REGULARIZATION_CONSTANT: 0.001,
+        # The scale of how important is to minimize the maximum similarity
+        # between embeddings of different labels,
+        # used only if 'loss_type' is set to 'margin'.
+        NEGATIVE_MARGIN_SCALE: 0.8,
+        # Dropout rate for embedding layers of dialogue features.
+        DROP_RATE_DIALOGUE: 0.1,
+        # Dropout rate for embedding layers of utterance level features.
+        DROP_RATE: 0.0,
+        # Dropout rate for embedding layers of label, e.g. action, features.
+        DROP_RATE_LABEL: 0.0,
+        # Dropout rate for attention.
+        DROP_RATE_ATTENTION: 0,
+        # Sparsity of the weights in dense layers
+        WEIGHT_SPARSITY: 0.8,
+        # ## Evaluation parameters
+        # How often calculate validation accuracy.
+        # Small values may hurt performance, e.g. model accuracy.
+        EVAL_NUM_EPOCHS: 20,
+        # How many examples to use for hold out validation set
+        # Large values may hurt performance, e.g. model accuracy.
+        EVAL_NUM_EXAMPLES: 0,
+        # If you want to use tensorboard to visualize training and validation metrics,
+        # set this option to a valid output directory.
+        TENSORBOARD_LOG_DIR: None,
+        # Define when training metrics for tensorboard should be logged.
+        # Either after every epoch or for every training step.
+        # Valid values: 'epoch' and 'minibatch'
+        TENSORBOARD_LOG_LEVEL: "epoch",
+        # Perform model checkpointing
+        CHECKPOINT_MODEL: False,
+        # Specify what features to use as sequence and sentence features.
+        # By default all features in the pipeline are used.
+        FEATURIZERS: [],
+        # If set to true, entities are predicted in user utterances.
+        ENTITY_RECOGNITION: False,
+        # List of intents to ignore
+        IGNORE_INTENTS_LIST: [],
+    }
+
+    def __init__(
+        self,
+        featurizer: Optional[TrackerFeaturizer] = None,
+        priority: int = UNLIKELY_INTENT_POLICY_PRIORITY,
+        max_history: Optional[int] = None,
+        model: Optional[RasaModel] = None,
+        fake_features: Optional[Dict[Text, List["Features"]]] = None,
+        entity_tag_specs: Optional[List[EntityTagSpec]] = None,
+        should_finetune: bool = False,
+        intent_thresholds: Dict[int, float] = None,
```

Comment:
Why not `List[float]` or `np.array`?

Reply:
Dictionary was easier because then the code does not rely on a particular order of intents in the threshold array. We still do rely on the order of intents from the domain to construct the intent label id though.